lib/charms/flume.py

```diff
@@ -1,129 +0,0 @@
-                              destination=self.dist_config.path('flume'),
-                              skip_top_level=True)
-        self.dist_config.add_users()
-        self.dist_config.add_dirs()
-        self.setup_flume_config()
-        unitdata.kv().set('flume_hdfs.installed', True)
-
-
-    def setup_flume_config(self):
-        '''
-        copy the default configuration files to flume_conf property
-        defined in dist.yaml
-        '''
-        default_conf = self.dist_config.path('flume') / 'conf'
-        flume_conf = self.dist_config.path('flume_conf')
-        flume_conf.rmtree_p()
-        default_conf.copytree(flume_conf)
-        # Now remove the conf included in the tarball and symlink our real conf
-        default_conf.rmtree_p()
-        flume_conf.symlink(default_conf)
-
-        flume_env = self.dist_config.path('flume_conf') / 'flume-env.sh'
-        if not flume_env.exists():
-            (self.dist_config.path('flume_conf') / 'flume-env.sh.template').copy(flume_env)
-
-        flume_conf = self.dist_config.path('flume_conf') / 'flume.conf'
-        if not flume_conf.exists():
-            (self.dist_config.path('flume_conf') / 'flume-conf.properties.template').copy(flume_conf)
-
-        flume_log4j = self.dist_config.path('flume_conf') / 'log4j.properties'
-        utils.re_edit_in_place(flume_log4j, {
-            r'^flume.log.dir.*': 'flume.log.dir={}'.format(self.dist_config.path('flume_logs')),
-        })        
-
-    def configure_flume(self):
-        config = hookenv.config()        
-        templating.render(
-            source='flume.conf.j2',
-            target=self.dist_config.path('flume_conf') / 'flume.conf',
-            context={'dist_config': self.dist_config, 'config': config})
-
-        flume_bin = self.dist_config.path('flume') / 'bin'
-        with utils.environment_edit_in_place('/etc/environment') as env:
-            if flume_bin not in env['PATH']:
-                env['PATH'] = ':'.join([env['PATH'], flume_bin])
-            env['FLUME_CONF_DIR'] = self.dist_config.path('flume_conf')
-            env['FLUME_CLASSPATH'] = self.dist_config.path('flume') / 'lib'
-            env['FLUME_HOME'] = self.dist_config.path('flume')
-
-        # flume_env = self.dist_config.path('flume_conf') / 'flume-env.sh'
-        # utils.re_edit_in_place(flume_env, {
-        # })
-        utils.run_as('flume', 'hdfs', 'dfs', '-mkdir', '-p', '/user/flume')
-
-    def run_bg(self, user, command, *args):
-        """
-        Start a Flume agent as the given user in the background.
-
-        :param str user: User to run flume agent
-        :param str command: Command to run
-        :param list args: Additional args to pass to the command
-        """
-        parts = [command] + list(args)
-        quoted = ' '.join("'%s'" % p for p in parts)
-        # This is here to force explicit execution on the background. Too much output causes Popen to fail. 
```

Comment:
The update should resolve this by redirecting the output to the file and including the `&`.